sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -417,10 +417,23 @@ private static Exception instantiateUnexpectedException(final UnexpectedExceptio
         final SwaggerMethodParser methodParser, RequestOptions options) {
         final int responseStatusCode = decodedResponse.getSourceResponse().getStatusCode();
 
+        boolean requestOptionsNoThrow = false;
+        boolean requestOptionsThrow = false;
+        if (options != null) {
+            requestOptionsNoThrow = options.getErrorOptions().contains(ErrorOptions.NO_THROW);
+            requestOptionsThrow = options.getErrorOptions().contains(ErrorOptions.THROW);
+        }
+
+        if (requestOptionsNoThrow && requestOptionsThrow) {
```

Comment:
can we check validity in the `RequestOptions.setErrorOptions` instead? I would simplify  getting options (i.e. we can only get `options.getErrorOptions().contains(ErrorOptions.NO_THROW)` and never check THROW

Reply:
Agreed

Reply:
That's a good question, my concern with `setErrorOptions` validating is as we add more options there could be more complex cases where having both could be permissible.

@JonathanGiles any thoughts?

Reply:
For now we can probably centralise the validation in the setter, and spread it out into appropriate places if complex cases arise.